src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs

```diff
@@ -37,31 +40,49 @@ public class Frame<TContext> : Frame
 
                     while (!_requestProcessingStopping && TakeStartLine(SocketInput) != RequestLineStatus.Done)
                     {
-                        if (SocketInput.CheckFinOrThrow())
+                        try
                         {
-                            // We need to attempt to consume start lines and headers even after
-                            // SocketInput.RemoteIntakeFin is set to true to ensure we don't close a
-                            // connection without giving the application a chance to respond to a request
-                            // sent immediately before the a FIN from the client.
-                            var requestLineStatus = TakeStartLine(SocketInput);
-
-                            if (requestLineStatus == RequestLineStatus.Empty)
+                            if (SocketInput.CheckFinOrThrow())
                             {
-                                return;
-                            }
+                                // We need to attempt to consume start lines and headers even after
+                                // SocketInput.RemoteIntakeFin is set to true to ensure we don't close a
+                                // connection without giving the application a chance to respond to a request
+                                // sent immediately before the a FIN from the client.
+                                var requestLineStatus = TakeStartLine(SocketInput);
+
+                                if (requestLineStatus == RequestLineStatus.Empty)
+                                {
+                                    return;
+                                }
 
-                            if (requestLineStatus != RequestLineStatus.Done)
+                                if (requestLineStatus != RequestLineStatus.Done)
+                                {
+                                    RejectRequest(RequestRejectionReason.InvalidRequestLine, requestLineStatus.ToString());
+                                }
+
+                                break;
+                            }
+                            await SocketInput;
+                        }
+                        catch (IOException ex)
+                        {
+                            // Silence ECONNRESET errors made between requests
+                            if (_requestLineStatus == RequestLineStatus.Empty &&
```

Comment:
Place `requestLineStatus` in an outer scope - better than making it a `Frame` member.